charm/Makefile

```diff
@@ -0,0 +1,140 @@
+BASENODE_DIR = $(CHARM_BUILD_DIR)/exec.d/
+BASENODE = $(BASENODE_DIR)/.done
+
+$(BUILDDIR) $(TMPDIR):
+	mkdir -p $@
+
+$(BASENODE_CACHE): | $(TMPDIR)
+	bzr branch $(BASENODE_REPO) $(BASENODE_CACHE_DIR)
+	touch $@
+
+.PHONY: $(BASENODE)
+$(BASENODE): | $(BASENODE_CACHE)
+	mkdir -p $(BASENODE_DIR)
+	cp -a $(BASENODE_CACHE_DIR) $(BASENODE_DIR)
+	touch $@
+
+$(CODETREE):
+	sudo snap install codetree --classic
+
+charm-dependencies: $(CHARM_DEPS)
+
+CHARM_DEPS = $(LAYER_PATH)/.done $(INTERFACE_PATH)/.done $(CHARM_WHEELS_DIR)/.done
+$(CHARM_DEPS): $(CHARM_SRC)/dependencies.txt $(CODETREE) | $(BASENODE_CACHE) $(TMPDIR)
+	cd $(TMPDIR) && $(CODETREE) $<
+	touch $(CHARM_DEPS)
+
+.PHONY: $(CHARM_BUILD_DIR)
+$(CHARM_BUILD_DIR): $(CHARM_DEPS) | $(BUILDDIR)
+	rm -rf $(CHARM_BUILD_DIR)
+	PIP_NO_INDEX=true PIP_FIND_LINKS=$(CHARM_WHEELS_DIR) charm build -o $(BUILDDIR) -s $(CHARM_SERIES) -n $(CHARM_NAME) ./charm/serial-vault
+
+/snap/bin/charm:
+	sudo snap install charm --classic
+
+$(CHARM_REPO_DIR)/.git:
+	git clone $(CHARM_PUBLISH_REPO) $(CHARM_REPO_DIR)
+
+charm-build-repo: | $(CHARM_REPO_DIR)/.git
+	cd $(CHARM_REPO_DIR) && git pull
+
+charm-build: /snap/bin/charm $(CHARM_BUILD_DIR) $(BASENODE)
+
+charm-collect: $(CHARM_DEPS)
+
+charm-clean:
+	rm -rf $(LAYER_PATH) $(INTERFACE_PATH) $(CHARM_WHEELS_DIR) $(TMPDIR)
+
+charm-deploy: bundle.yaml
+	@echo "Deploying $(CHARM_NAME)..."
+	juju deploy ./charm/bundle.yaml
+
+charm-upgrade:
+	juju upgrade-charm $(CHARM_NAME) --path $(CHARM_BUILD_DIR)
+
+charm-check: IP=$(shell juju status $(CHARM_NAME) --format=short |tail -1| awk '{print $$3}')
+charm-check: PROXY=$(shell juju status haproxy --format=short |tail -1| awk '{print $$3}')
+charm-check: PORT=$(shell grep port: charm/layer.yaml | awk '{print $$2}')
+charm-check:
+	@echo Checking $(CHARM_NAME) unit...
+	curl http://$(IP):$(PORT)/_status/ping
+	curl http://$(IP):$(PORT)/_status/check
+	@echo Checking haproxy routing unit...
+	curl http://$(PROXY)/_status/ping
+	curl http://$(PROXY)/_status/check
+
+
+charm-publish: TRUNK_REVNO=$(shell bzr revision-info)
+charm-publish: charm-build charm-build-repo
+	rsync -a -m --ignore-times --exclude .git --delete $(CHARM_BUILD_DIR)/ $(CHARM_REPO_DIR)/
+	cd $(CHARM_REPO_DIR) && git add --no-ignore-removal .
+	# Need to force username and email because git can't autodetect this
+	# on jenkaas slaves.
+	cd $(CHARM_REPO_DIR) && git -c user.name="$(GIT_USERNAME)" -c user.email="$(GIT_EMAIL)"  commit -am "Build of $(CHARM_NAME) from $(TRUNK_REVNO)"
+	cd $(CHARM_REPO_DIR) && git tag "$(shell git rev-parse --short HEAD)"
+	cd $(CHARM_REPO_DIR) && git push origin master --tags
+
+charm-private-publish:
+	git ls-remote -q $(PRIVATE_PUBLISH_REPO) || ( cd $$(mktemp -d); git init; git commit --allow-empty -m 'init'; git push $(PRIVATE_PUBLISH_REPO) master)
+	$(MAKE) charm-publish CHARM_PUBLISH_REPO=$(PRIVATE_PUBLISH_REPO) CHARM_REPO_DIR=$(PRIVATE_CHARM_REPO_DIR)
+
+# get swift credentials
+-include ~/.config/swift/serial-vault
```

Comment:
We need to have this file with swift credentials in the `serial-vault` environment